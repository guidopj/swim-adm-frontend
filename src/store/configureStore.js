import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import redirectMiddleware from 'middlewares/redirectMiddleware'
import history from 'history.js'
import { routerMiddleware } from 'connected-react-router'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

import rootSaga from 'sagas';
import rootReducer from 'reducers/rootReducer';

const persistConfig = {
	key: 'root',
	storage,
  }

  const mainReducer = rootReducer(history)
  const persistedReducer = persistReducer(persistConfig, mainReducer)

/** Saga Middleware */
const sagaMiddleware = createSagaMiddleware();

let middlewares = applyMiddleware(sagaMiddleware, routerMiddleware(history), redirectMiddleware);


export default function configureStore (initialState) {
	const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(persistedReducer,initialState, composeEnhancers(
        middlewares,
	  )
	);
	let persistor = persistStore(store)
	
	sagaMiddleware.run(rootSaga);
    return { store, persistor }
}

/* export const sagaMiddleware = createSagaMiddleware()

const composeStore = compose(
	applyMiddleware(
		redirectMiddleware,
		metricsMiddleware,
		sagaMiddleware,
		generalMiddleware,
		postMessageMiddleware
	),
	window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore)

const store = composeStore(reducers)

sagaMiddleware.run(rootSaga)

export default store */