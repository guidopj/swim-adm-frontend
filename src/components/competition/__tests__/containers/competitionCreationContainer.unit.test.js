import React from 'react';
import CompetitionCreationContainer from 'components/competition/containers/competitionCreationContainer';
import configureStore from 'store/configureStore'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'
import history from 'history.js'
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe ('Competition Creation Container', () => {
  const CreateNewCompetition = jest.fn()
  const { store } = configureStore({})

  it('renders correctly', () => {

    const location = {
      state: {
        competitionName: 'a',
      }
    }

    const tree = shallow
      (
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <CompetitionCreationContainer 
              actions={{
                CreateNewCompetition,
              }}
              location={location}
            />
          </ConnectedRouter>
        </Provider>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })
})