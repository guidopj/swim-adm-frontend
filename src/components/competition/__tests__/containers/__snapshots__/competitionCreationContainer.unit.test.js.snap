// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competition Creation Container renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(CompetitionCreationContainer)
      actions={
        Object {
          "CreateNewCompetition": [MockFunction],
        }
      }
      location={
        Object {
          "state": Object {
            "competitionName": "a",
          },
        }
      }
    />
  </Connect(ConnectedRouterWithContext)>
</ContextProvider>
`;
